rust/src/connection.rs

```diff
@@ -0,0 +1,56 @@
+use crate::config;
+use crate::wallet;
+use anyhow::Result;
+
+/// blocks the current thread, if you want this to be run asynchronously you need to spawn a thread
+/// from the outside.
+pub async fn keep_alive() -> Result<()> {
+    let mut connected = false;
+    loop {
+        let alive = check_alive().await?;
+
+        if !connected || !alive {
+            connected = connect().await?;
+        }
+
+        // looping here indefinitely to keep the connection with the maker alive.
+        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
+    }
+}
+
+async fn check_alive() -> Result<bool> {
+    tracing::trace!("Checking if maker is alive");
+    let client = reqwest::Client::builder()
+        .timeout(config::TCP_TIMEOUT)
+        .build()?;
```

Comment:
🔧 Would it be safe to `expect` here? It would be nice if `check_alive` were infallible.

Reply:
> 🔧 Would it be safe to `expect` here? It would be nice if `check_alive` were infallible.

it should be, we're passing static arguments after all into the builder.

Reply:
thanks for the feedback, I adapted the implementation, but figured the improvements were not worthwhile the additional round of testing. Also giving that this is very much still a workaround patch.